tap_deputy/client.py

```diff
@@ -42,25 +85,27 @@ def refresh(self):
                 'scope': 'longlife_refresh_token'
             })
 
-        self.__config['refresh_token'] = data['refresh_token']
-
-        with open(self.__config_path, 'w') as file:
-            json.dump(self.__config, file, indent=2)
-
         self.__refresh_token = data['refresh_token']
         self.__access_token = data['access_token']
+        # pad by 10 seconds for clock drift
+        self.__expires_at = now() + timedelta(seconds=data['expires_in'] - 10)
+
+        if not self.__dev_mode:
+            self.__config['refresh_token'] = self.__refresh_token
+            self.__config['access_token'] = self.__access_token
+            self.__config['expires_at'] = strftime(self.__expires_at)
```

Comment:
I've been pushing for this since the tap is currently in a state where changing the config elsewhere can cause a side effect in this update to the file on disk (since `self.__config` references the same object that the external scopes have access to, there are no guarantees here).

The writing of the new config values should be
1. json.load current config on disk
2. Update the value
3. json.dump the updated config to disk

That way no side effects can occur since it's a transactional operation at that point.

Reply:
@dmosorast I have update the implementation as per your suggestion.

Reply:
Updated the implementation as per your suggestion and also removed the `self.__config ` class variable since we are now reading the config from the disk itself before writing it.